src/ray/protobuf/gcs.proto

```diff
@@ -200,6 +201,70 @@ message ProfileTableData {
   repeated ProfileEvent profile_events = 4;
 }
 
+message ProfileEventEntry {
+  // The start timestamp of the event time.
+  int64 start_time = 1;
+  // The end timestamp of the event. Empty if it's a point event, e.g. TaskStatus
+  // change.
+  int64 end_time = 2;
+  // Additional data associated with the event. This data must be serialized
+  // using JSON.
+  string extra_data = 3;
+  // Customized event name if not a TaskStatus change point event.
+  string event_name = 4;
+}
+
+message ProfileEvents {
+  // The type of the component that generated the event, e.g., worker or
+  // object_manager, or node_manager.
+  string component_type = 1;
+  // An identifier for the component that generated the event.
+  bytes component_id = 2;
+  // Node IP address.
+  string node_ip_address = 3;
+  // Events.
+  repeated ProfileEventEntry events = 4;
+}
+
+// Represents task states which could be changed during task execution.
+message TaskStateUpdate {
+  // Node that runs the task.
+  bytes node_id = 1;
+  // Timestamp when status changes to PENDING_ARGS_AVAIL.
+  int64 pending_args_avail_ts = 2;
+  // Timestamp when status changes to PENDING_NODE_ASSIGNEMENT.
+  int64 pending_node_assignement_ts = 3;
+  // Timestamp when status changes to SUBMITTED_TO_WORKER.
+  int64 submitted_to_worker_ts = 4;
+  // Timestamp when status changes to RUNNING.
+  int64 running_ts = 5;
+  // Timestamp when status changes to FINISHED.
+  int64 finished_ts = 6;
+  // Timestamp when status changes to FAILED.
+  int64 failed_ts = 7;
+}
+
+// Represents events and state changes from a single task run.
+message TaskEvents {
+  // Metadata shared by all event types.
+  bytes task_id = 1;
+  // The current retry attempt number for the task.
+  int32 attempt_number = 2;
+  // Task Spec.
+  TaskSpec task_spec = 3;
```

Comment:
Maybe we can just send TaskInfo here? TaskSpec includes the objects inlined iirc, so it will be very expensive.

Reply:
Yeah - but then we will need to translate  the TaskSpec into`TaskInfo`? I think I could try benchmarking it.

Reply:
I think at least we need to reset inlined objects. As long as that's not reported whatever is fine for me